phygnn/model_interfaces/base_model.py

```diff
@@ -427,6 +463,33 @@ def _get_item_number(arr):
 
         return n
 
+    @staticmethod
+    def make_one_hot_feature_names(feature_names, one_hot_categories):
+        """
+        Update feature_names after one-hot encoding
+
+        Parameters
+        ----------
+        feature_names : list
+            Input feature names
+        one_hot_categories : dict
+            Features to one-hot encode using given categories
+
+        Returns
+        -------
+        feature_names : list
+            Updated list of feature names with one_hot categories
+        """
+        for name, categories in one_hot_categories.items():
+            if name in feature_names:
+                feature_names.remove(name)
+
+            for c in categories:
+                if c not in feature_names:
+                    feature_names.append(c)
```

Comment:
How about using `f'{name}_{c}'` or similar for the new feature names to reduce the possibility of feature name collisions and simplify debugging?

Reply:
Fine by me, we should also update PreProcessing for consistency: https://github.com/NREL/phygnn/blob/3345f50aca33521e893abf587ff62a90a924223b/phygnn/utilities/pre_processing.py#L226-L285
@grantbuster?

Reply:
What about just raising an error if one of the OHE categorical values matches a pre-existing feature name? I think its really nice being able to specify the OHE categories and know exactly what the new columns are going to be labeled.

Reply:
added a check_one_hot_categories method to PreProcessing that checks for duplicate categories, and then if you provide the feature_names:
1) Checks to ensure the one-hot features (keys) are in the list of feature_names
2) Checks to ensure the one-hot categories aren't in the feature_names list